### PR TITLE
Fix alert plugin instance filter (#7172)

### LINKED ISSUE #7209: [Bug] [api-service] Filter Security: warning-instance failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

When I filter `warning-instance` object, I found the result is not expect, and the result not filter
image

### What you expected to happen

Filter base on I enter

### How to reproduce

Go to `http://<IP>:<PORT>/dolphinscheduler/ui/#/security/warning-instance` and see whether it could filter or not.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhongjiajie**
alert plugin could not filter by given
pattern, cause api server do not handle
parameter searchVal, This patch add parameter
`searchVal` for alert plugin.

fix: #7209

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/AlertPluginInstanceServiceImpl.java [hunk 1]
```diff
@@ -188,14 +188,20 @@ public boolean checkExistPluginInstanceName(String pluginInstanceName) {
...
+            putMsg(result,Status.USER_NO_OPERATION_PERM);
+            return result;
+        }
+
+        Page<AlertPluginInstance> page = new Page<>(pageNo, pageSize);
+        IPage<AlertPluginInstance> alertPluginInstanceIPage = alertPluginInstanceMapper.queryByInstanceNamePage(page, searchVal);
+
+        PageInfo<AlertPluginInstance> pageInfo = new PageInfo<>(pageNo, pageSize);
+        pageInfo.setTotal((int) alertPluginInstanceIPage.getTotal());
+        pageInfo.setTotalList(alertPluginInstanceIPage.getRecords());
```
Comment (CalvinKirs) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/AlertPluginInstanceServiceImpl.java]: You deleted the relevant UI conversion code

Comment (zhongjiajie) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/AlertPluginInstanceServiceImpl.java]: Delete what? 👀

Comment (CalvinKirs) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/AlertPluginInstanceServiceImpl.java]: buildPluginInstanceVOList

Comment (zhongjiajie) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/AlertPluginInstanceServiceImpl.java]: Oh, I got your point 👍  forgive my poor Java

Comment (zhongjiajie) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/AlertPluginInstanceServiceImpl.java]: I add this

**Comment by zhongjiajie** [comment]
e2e CI failed, rebase and force push to re-run it

**Comment by zhongjiajie** [comment]
Still failed e2e and it is odd that show `noSuchEelmet` in L162

https://github.com/apache/dolphinscheduler/blob/37ba1eb5adb962f6f13825ee0375e82b4823f254/dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/WorkflowE2ETest.java#L159-L162

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Comment by zhongjiajie** [comment]
Thanks for merging.

**Pull request merged**